include/benchmark/benchmark.h

```diff
@@ -1442,13 +1459,14 @@ class Fixture : public internal::Benchmark {
 #define BENCHMARK_PRIVATE_DECLARE(n)                                           \
   /* NOLINTNEXTLINE(misc-use-anonymous-namespace) */                           \
   static ::benchmark::internal::Benchmark const* const BENCHMARK_PRIVATE_NAME( \
-      n) [[maybe_unused]]
+      n) BENCHMARK_UNUSED
 
 #define BENCHMARK(...)                                                \
   BENCHMARK_PRIVATE_DECLARE(_benchmark_) =                            \
       (::benchmark::internal::RegisterBenchmarkInternal(              \
-          std::make_unique<::benchmark::internal::FunctionBenchmark>( \
-              #__VA_ARGS__, __VA_ARGS__)))
+          benchmark::internal::make_unique<                           \
+              ::benchmark::internal::FunctionBenchmark>(#__VA_ARGS__, \
+                                                        __VA_ARGS__)))
```

Comment:
I think there's a bug here: the macro should call `::benchmark::internal::make_unique` not `benchmark::internal::make_unique`.

We're now getting [compilation failures](https://buildkite.com/apache-arrow/arrow-bci-benchmark-on-amd64-c6a-4xlarge-linux/builds/937#0195d209-e346-4c84-8895-b82b61d1a899/95-2687) because of this, example:
```
/var/lib/buildkite-agent/builds/amd64-c6a-4xlarge-linux-i-0a326a82268fb1cf0-1/apache-arrow/arrow-bci-benchmark-on-amd64-c6a-4xlarge-linux/arrow/cpp/src/parquet/page_index_benchmark.cc:60:1: error: 'parquet::benchmark::internal' has not been declared
   60 | BENCHMARK(BM_ReadOffsetIndex)->Apply(PageIndexSetArgs);
      | ^~~~~~~~~
```